src/multibody/geometry.hxx

```diff
@@ -31,30 +31,48 @@
 
 namespace se3
 {
+  inline GeometryData::GeometryData(const GeometryModel & modelGeom)
+    : oMg(modelGeom.ngeoms)
+
+#ifdef WITH_HPP_FCL   
+    , activeCollisionPairs(modelGeom.collisionPairs.size(), true)
+    , distanceRequest (true, 0, 0, fcl::GST_INDEP)
+    , distanceResults(modelGeom.collisionPairs.size())
+    , collisionRequest (1, false, false, 1, false, true, fcl::GST_INDEP)
+    , collisionResults(modelGeom.collisionPairs.size())
+    , radius()
+    , collisionPairIndex(-1)
+    , innerObjects()
+    , outerObjects()
+  {
+    collisionObjects.reserve(modelGeom.geometryObjects.size());
+    BOOST_FOREACH( const GeometryObject & geom, modelGeom.geometryObjects)
+      { collisionObjects.push_back
+          (fcl::CollisionObject(geom.fcl)); }
+    fillInnerOuterObjectMaps(modelGeom);
+  }
+#else
+  {}
+#endif // WITH_HPP_FCL   
 
-  inline GeomIndex GeometryModel::addGeometryObject(const GeometryObject& object)
+  inline GeomIndex GeometryModel::addGeometryObject(GeometryObject object,
+                                                    const Model & model)
   {
-    Index idx = (Index) (ngeoms ++);
+    assert( (object.parentFrame != -1) || (object.parentJoint != -1) );
+
+    if( object.parentJoint == -1 )  // if no parent joint, autofill the attribute.
+      object.parentJoint = model.frames[object.parentFrame].parent; 
+
+    assert( (object.parentFrame == -1) 
+            || (model.frames[object.parentFrame].type == se3::BODY)  );
+    assert( (object.parentFrame == -1) 
+            || (model.frames[object.parentFrame].parentJoint == object.parentJoint) );
```

Comment:
I remember I said it could be -1 as it is not a necessary information. I
guess I forgot it is not signed. Sorry about that.

There are a few places where I used int instead of FrameIndex because of
this. It's ugly. I think we should add a typedef SignedIndex. Let's keep
this for later.

For now, we can simply comment the automatic detection with a todo to
remind us to re enable it when relevant.